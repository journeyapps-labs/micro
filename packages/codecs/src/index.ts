export * from './parsers';
export * from './codecs';
