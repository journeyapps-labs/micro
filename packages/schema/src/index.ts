export * from './json-schema/keywords';
export * from './json-schema/parser';

export * from './definitions';
export * from './validators';
export * from './utils';
