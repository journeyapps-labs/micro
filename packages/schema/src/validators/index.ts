export * from './ts-codec-validator';
export * from './schema-validator';
export * from './zod-validator';
