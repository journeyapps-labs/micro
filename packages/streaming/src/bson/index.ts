export * from "./buffer-array";
export * from "./constants";
export * from "./encoder";
export * from "./decoder";
export * from "./header";
