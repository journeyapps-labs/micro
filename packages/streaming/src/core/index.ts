export * from './backpressure';
export * from './transformers';
export * from './node-utils';
export * from './utils';
