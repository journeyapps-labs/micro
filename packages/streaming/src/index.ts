export * as compat from './core/cross-stream';
export * as middleware from './middleware';
export * as bson from './bson';
export * from './core';
