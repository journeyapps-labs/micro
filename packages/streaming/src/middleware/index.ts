export * from './streamed-body-parser-v1';
export * from './streamed-body-parser-v2';
