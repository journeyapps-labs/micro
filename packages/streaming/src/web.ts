export * as compat from "./core/cross-stream";
export * as bson from "./bson";

export * from "./core/transformers";
export * from "./core/utils";
