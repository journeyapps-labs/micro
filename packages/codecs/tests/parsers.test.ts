import { describe, it, expect } from "vitest";

import * as codecs from "../src";
import * as t from "ts-codec";

const generate = (codec: t.AnyCodec) => {
  const encoded = t.generateJSONSchema(codec, {
    parsers: codecs.parsers,
    target: t.TransformTarget.Encoded,
  });
  const decoded = t.generateJSONSchema(codec, {
    parsers: codecs.parsers,
    target: t.TransformTarget.Decoded,
  });

  return { encoded, decoded };
};

describe("parsers", () => {
  it("should correctly generate date schemas", () => {
    expect(generate(codecs.date)).toMatchSnapshot();
  });

  it("should correctly generate buffer schemas", () => {
    expect(generate(codecs.buffer)).toMatchSnapshot();
  });

  it("should correctly generate ObjectId schemas", () => {
    expect(generate(codecs.ObjectId)).toMatchSnapshot();
  });

  it("should correctly generate ResourceId schemas", () => {
    expect(generate(codecs.ResourceId)).toMatchSnapshot();
  });

  it("should correctly generate Filterable schema from Object", () => {
    const filterObject1 = t.object({
      test1: t.string,
    });

    const filterObject2 = t.object({
      test2: t.number,
    });

    const filterObject3 = t.object({
      test1: t.string,
    });

    const filterObject4 = filterObject1.and(filterObject2).and(filterObject3);

    const filterableObject = codecs.FilterProperties(filterObject4);
    expect(generate(filterableObject)).toMatchSnapshot();
  });
});
