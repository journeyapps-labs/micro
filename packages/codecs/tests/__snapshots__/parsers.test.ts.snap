// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`parsers > should correctly generate Filterable schema from Object 1`] = `
{
  "decoded": {
    "additionalProperties": false,
    "definitions": {},
    "properties": {
      "test1": {
        "anyOf": [
          {
            "type": "string",
          },
          {
            "items": {
              "type": "string",
            },
            "type": "array",
          },
          {
            "additionalProperties": false,
            "properties": {
              "exists": {
                "type": "boolean",
              },
            },
            "required": [
              "exists",
            ],
            "type": "object",
          },
        ],
      },
      "test2": {
        "anyOf": [
          {
            "type": "number",
          },
          {
            "items": {
              "type": "number",
            },
            "type": "array",
          },
          {
            "additionalProperties": false,
            "properties": {
              "exists": {
                "type": "boolean",
              },
            },
            "required": [
              "exists",
            ],
            "type": "object",
          },
        ],
      },
    },
    "required": [],
    "type": "object",
  },
  "encoded": {
    "additionalProperties": false,
    "definitions": {},
    "properties": {
      "test1": {
        "anyOf": [
          {
            "type": "string",
          },
          {
            "items": {
              "type": "string",
            },
            "type": "array",
          },
          {
            "additionalProperties": false,
            "properties": {
              "exists": {
                "type": "boolean",
              },
            },
            "required": [
              "exists",
            ],
            "type": "object",
          },
        ],
      },
      "test2": {
        "anyOf": [
          {
            "type": "number",
          },
          {
            "items": {
              "type": "number",
            },
            "type": "array",
          },
          {
            "additionalProperties": false,
            "properties": {
              "exists": {
                "type": "boolean",
              },
            },
            "required": [
              "exists",
            ],
            "type": "object",
          },
        ],
      },
    },
    "required": [],
    "type": "object",
  },
}
`;

exports[`parsers > should correctly generate ObjectId schemas 1`] = `
{
  "decoded": {
    "bsonType": "ObjectId",
    "definitions": {},
  },
  "encoded": {
    "definitions": {},
    "type": "string",
  },
}
`;

exports[`parsers > should correctly generate ResourceId schemas 1`] = `
{
  "decoded": {
    "definitions": {},
    "properties": {
      "_id": {
        "bsonType": "ObjectId",
      },
    },
    "required": [
      "_id",
    ],
    "type": "object",
  },
  "encoded": {
    "definitions": {},
    "properties": {
      "id": {
        "type": "string",
      },
    },
    "required": [
      "id",
    ],
    "type": "object",
  },
}
`;

exports[`parsers > should correctly generate buffer schemas 1`] = `
{
  "decoded": {
    "definitions": {},
    "nodeType": "buffer",
  },
  "encoded": {
    "definitions": {},
    "type": "string",
  },
}
`;

exports[`parsers > should correctly generate date schemas 1`] = `
{
  "decoded": {
    "definitions": {},
    "nodeType": "date",
  },
  "encoded": {
    "definitions": {},
    "type": "string",
  },
}
`;
